baggage.go

```diff
@@ -0,0 +1,113 @@
+package pyroscope
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"go.opentelemetry.io/otel/baggage"
+)
+
+// FilterFunc returns true if this key should be used.
+type FilterFunc func(key string) bool
+
+// TransformFunc transforms the key.
+type TransformFunc func(key string) string
+
+// BaggageConfig contains configuration options for filtering and transforming
+// baggage members.
+type BaggageConfig struct {
+	filters    []FilterFunc
+	transforms []TransformFunc
+}
+
+// BaggageOption sets a option on a BaggageConfig.
+type BaggageOption func(config *BaggageConfig)
+
+// WithFilters sets filtering functions to apply to the baggage.
+func WithFilters(filters ...FilterFunc) BaggageOption {
+	return func(config *BaggageConfig) {
+		config.filters = append(config.filters, filters...)
+	}
+}
+
+// WithTransforms sets transformation functions to apply to the baggage.
+func WithTransforms(transforms ...TransformFunc) BaggageOption {
+	return func(config *BaggageConfig) {
+		config.transforms = append(config.transforms, transforms...)
+	}
+}
+
+// K6Options provides default options to select k6 members from the baggage.
+func K6Options() []BaggageOption {
```

Comment:
Should we perhaps place the newly added functionality into a submodule? For example, contrib/k6? There are a number of reasons why this could be beneficial:
 1. This is an experimental feature, and the API may change. It would be better if we had clear boundaries between stable and experimental features.
 2. The API surface of the main module would be cleaner: I expect that only a very small fraction of users are interested in K6Options and baggage handling.
 3. We intentionally keep the number of dependencies of the main module low. otel/baggage dependency is quite heavy.

Reply:
I agree with this 👍